go/vt/vtgate/endtoend/vschema/vschema_test.go

```diff
@@ -0,0 +1,169 @@
+	tabletHostName = flag.String("tablet_hostname", "", "the tablet hostname")
+
+	schema = `
+create table vt_user (
+	id bigint,
+	name varchar(64),
+	primary key (id)
+) Engine=InnoDB;
+	
+create table main (
+	id bigint,
+	val varchar(128),
+	primary key(id)
+) Engine=InnoDB;
+`
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	exitCode := func() int {
+		var cfg vttest.Config
+		cfg.Topology = &vttestpb.VTTestTopology{
+			Keyspaces: []*vttestpb.Keyspace{{
+				Name: "ks",
+				Shards: []*vttestpb.Shard{{
+					Name: "80",
+				}},
+			}},
+		}
+		cfg.ExtraMyCnf = []string{path.Join(os.Getenv("VTTOP"), "config/mycnf/rbr.cnf")}
+		if err := cfg.InitSchemas("ks", schema, nil); err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			os.RemoveAll(cfg.SchemaDir)
+			return 1
+		}
+		defer os.RemoveAll(cfg.SchemaDir)
+
+		cfg.TabletHostName = *tabletHostName
+
+		//List of users authorized to execute vschema ddl operations
+		cfg.ExtraArg = append(cfg.ExtraArg, "-vschema_ddl_authorized_users=%")
+
+		cluster = &vttest.LocalCluster{
+			Config: cfg,
+		}
+		if err := cluster.Setup(); err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			cluster.TearDown()
+			return 1
+		}
+		defer cluster.TearDown()
+
+		vtParams = mysql.ConnParams{
+			Host: "localhost",
+			Port: cluster.Env.PortForProtocol("vtcombo_mysql_port", ""),
+		}
+		mysqlParams = cluster.MySQLConnParams()
+		grpcAddress = fmt.Sprintf("localhost:%d", cluster.Env.PortForProtocol("vtcombo", "grpc"))
+
+		return m.Run()
+	}()
+	os.Exit(exitCode)
+}
+
+func TestVSchema(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	// Test the blank database with no vschema
+	exec(t, conn, "insert into vt_user (id,name) values(1,'test1'), (2,'test2'), (3,'test3'), (4,'test4')")
+	qr := exec(t, conn, "select id, name from vt_user order by id")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[INT64(1) VARCHAR("test1")] [INT64(2) VARCHAR("test2")] [INT64(3) VARCHAR("test3")] [INT64(4) VARCHAR("test4")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	qr = exec(t, conn, "delete from vt_user")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	//Test Blank VSCHEMA
+	qr = exec(t, conn, "SHOW VSCHEMA TABLES")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[VARCHAR("dual")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
+
+	//Use the DDL to create an unsharded vschema and test again
+
+	//Create VSchema
+	exec(t, conn, "begin")
+	exec(t, conn, "ALTER VSCHEMA ADD TABLE main")
+	exec(t, conn, "ALTER VSCHEMA ADD TABLE vt_user")
+	exec(t, conn, "commit")
+
+	//Test Showing Tables
+	qr = exec(t, conn, "SHOW VSCHEMA TABLES")
+	if got, want := fmt.Sprintf("%v", qr.Rows), `[[VARCHAR("dual")] [VARCHAR("main")] [VARCHAR("vt_user")]]`; got != want {
+		t.Errorf("select:\n%v want\n%v", got, want)
+	}
```

Comment:
@deepthi above test case fails intermittently and I am not able to figure out the reason for that. 
Sometimes none of the requested Vschema gets created and sometimes only one of them is present. 
Here is the screenshot when it fails. 

<img width="846" alt="Screen Shot 2019-10-18 at 1 53 20 PM" src="https://user-images.githubusercontent.com/1236152/67082434-c7351f00-f1b6-11e9-897a-e3f8f58015c2.png">

Reply:
@sougou is there some caching that needs to be disabled?

Reply:
I think this has to do with the delay of the vtcombo watching the SrvVSchema. You could put a sleep, but that may not completely eliminate flakyness.

You can instead check this in a loop for 10-20 seconds, and then fail.

Reply:
I tried looping and waiting for 60 seconds but that still not guaranteed for updated vschema. Then I tried to add a select statement after the vschema updated and looks like that solved it. The test cases are not failing. 

Can you please review again if that approach is fine?